### ISSUE TITLE
drp is not zero

### ISSUE THREAD

**Issue by londumas** [igmhub/SaclayMocks#11]
Currently we measure `drp_QSO = -1.174 +/- 0.039`, `drp_DLA = -1.193 +/- 0.627`. There are no obvious reasons for this parameter to be different than zero. It might be simply linked to random fluctuations of the velocity on mocks. We want to look at that with 10 realizations.

**Comment by jmarclegoff** [comment]
Hélion,
Do you really believe that drp_QSO = -1.174 +/- 0.039 can be a statistical fluctuation ?
Or you do not believe in the estimate of the error ?

**Comment by londumas** [comment]
@jmarclegoff, I don't know if you remember what we observed in your previous mocks (last year of my thesis),
but we also measured `drp!=0` in each realization, but the distribution was compatible with zero.
At that time, we concluded, through one of your plot where you looked at the velocity gradient along the z-axis, that it was linked to non vanishing large scale mode in the velocity gradient.
So maybe we have the same effect in the new mocks.

**Comment by jmarclegoff** [comment]
@londumas, now I think I remember. So let's wait for the additional realisations.
But what does it mean for real data ? The drp that we observe is meaningless or these non vanishing large scale mode in the velocity gradient exist in the mocks and not in the data ? 
I would need to think more about that.

**Comment by londumas** [comment]
This realization v4.2 has `drp_QSO = -0.94 +/- 0.0838`. Taking into account the `-100km/s` from quickquasar redshift systematic error. So the shift is indeed still there. Let's wait to see other realizations.

**Comment by TEtourneau** [comment]
Here is the measured drp on the 10 cross-correlations from the raw-mocks of version v4.4
![Screen Shot 2019-06-28 at 3 52 54 PM](https://user-images.githubusercontent.com/36950825/60349298-993a8d80-99c1-11e9-8965-5384dde7374a.png)

drp mean is -1.068 +/- 0.083 (1 sigma)

Edit: In fact it is from the eboss-0.2 (quickquasars + DLA) cross-correlations !
The measure mean drp on the raw-mocks is:
drp = -1.105 +/- 0.089

**Comment by londumas** [comment]
@TEtourneau, good thanks for the plot. Could you add a plot of the stack or at least one correlation, along the first slice at rt==0? Like in here https://github.com/igmhub/LyaCoLoRe/issues/41#issuecomment-475660138

**Comment by jmarclegoff** [comment]
Between versions 4.4 and 4.5, I corrected a stupid bug in the function that computes Gaussian smearing weights. This bug was changing where we read in the box by half a pixel, i.e. 1.095 Mpc/h.
I now realize that this should bring the mean drp to zero!
from -1.068 +- 0.083 to +0.027 +- 0.083 
To be checked with version 4.5 mocks.

**Comment by londumas** [comment]
@jmarclegoff, good. Looking at @TEtourneau plots, this can be checked with one realization.

**Comment by TEtourneau** [comment]
Here are the plots:
![Screen Shot 2019-06-28 at 5 17 47 PM](https://user-images.githubusercontent.com/36950825/60352726-c3437e00-99c8-11e9-9fbc-2873e9f66004.png)
![Screen Shot 2019-06-28 at 5 18 16 PM](https://user-images.githubusercontent.com/36950825/60352727-c3dc1480-99c8-11e9-9256-1cd0167ed5e6.png)

The cross-correlations are taken from the raw-mocks. (I only took 9 on the 10 realisations because I think one of the fits didn't converge)

**Comment by londumas** [comment]
@TEtourneau, Thanks for this. Indeed it is in the opposite direction as the London mocks. @jfarr03, maybe you have a similar bug as Saclay had?

**Comment by TEtourneau** [comment]
The shift in the London mocks is much smaller than our, so I don't think it could be something similar. But I guess it is always worth to check it :)

**Comment by jmarclegoff** [comment]
@TEtourneau, indeed have an effect of half a simulation voxels, 
@jfarr03 the effect for London could rather be something like half a spectrum pixel?

**Comment by jfarr03** [comment]
@jmarclegoff Yes in the London mocks drp is around 1/2 the cell width. Can you point me to the code you changed to correct the error?

**Comment by jmarclegoff** [comment]
@jfarr03 This is a stupid bug, from a time when I was learning python and fighting to vectorize the code so that I did not pay enough attention to the algorithm. I positioned the center of the cell at the edge of it! 
at line 53 of https://github.com/igmhub/SaclayMocks/blob/master/bin/make_spectra.py
        cell_center = sp.array([(ix+0.5)*DX-LX/2,(iy+0.5)*DY-LY/2,(iz+0.5)*DZ-LZ/2+R0])
used to be
        cell_center = sp.array([ix*DX-LX/2,iy*DY-LY/2,iz*DZ-LZ/2+R0])

**Comment by jvstermer** [comment]
For the latest version of the mocks, v4.5.0: 
![document-1](https://user-images.githubusercontent.com/39825957/60582318-db304e80-9d88-11e9-983d-82529d1b8bb0.png)
![document-2](https://user-images.githubusercontent.com/39825957/60582319-db304e80-9d88-11e9-9a54-c9b652e30a99.png)

From the cross correlation of raw mocks with a down sampling of 600 000 quasars. 
This result is to be confirmed with additional realizations of the mocks.

**Comment by londumas** [comment]
@jvstermer, this looks very good. Thanks.
